packages/eslint-plugin/src/rules/sort-ngmodule-metadata-arrays.ts

```diff
@@ -16,39 +16,106 @@ export default createESLintRule<Options, MessageIds>({
         'Ensures ASC alphabetical order for `NgModule` metadata arrays for easy visual scanning',
       category: 'Best Practices',
       recommended: false,
+      suggestion: true,
     },
     fixable: 'code',
     schema: [],
     messages: {
       sortNgmoduleMetadataArrays:
         '`NgModule` metadata arrays should be sorted in ASC alphabetical order',
+      suggestFix: 'Sort members (WARNING! comments will probably be messed up)',
     },
   },
   defaultOptions: [],
   create(context) {
+    const sourceCode = context.getSourceCode();
+
     return {
       [`${Selectors.MODULE_CLASS_DECORATOR} Property ArrayExpression`]({
         elements,
       }: TSESTree.ArrayExpression) {
-        const unorderedNodes = elements
-          .filter(TSESLintASTUtils.isIdentifier)
-          .map((current, index, list) => [current, list[index + 1]])
-          .find(([current, next]) => {
-            return next && current.name.localeCompare(next.name) === 1;
-          });
+        const [expressions, identifiers] = partition(
+          elements,
+          TSESLintASTUtils.isIdentifier,
+        );
+        const firstUnorderedNode = getFirstUnderedNode(identifiers);
 
-        if (!unorderedNodes) return;
+        if (!firstUnorderedNode) {
+          return;
+        }
 
-        const [unorderedNode, nextNode] = unorderedNodes;
+        const hasComments = elements.some((element) =>
+          ASTUtils.hasComments(sourceCode, element),
+        );
+        const fix: TSESLint.ReportFixFunction = (fixer) =>
+          getFixes(sourceCode, fixer, elements, expressions, identifiers);
         context.report({
-          node: nextNode,
+          node: firstUnorderedNode,
           messageId: 'sortNgmoduleMetadataArrays',
-          fix: (fixer) => [
-            fixer.replaceText(unorderedNode, nextNode.name),
-            fixer.replaceText(nextNode, unorderedNode.name),
-          ],
+          ...(hasComments
+            ? { suggest: [{ messageId: 'suggestFix', fix }] }
+            : { fix }),
```

Comment:
I even tried to use `fix`, even if there were comments, but besides being very costly to maintain such an implementation, it's also very prone to failure. 😞

Reply:
One thing that came to my mind right now, is:

we could add an option called something like `alwaysFix` or `ignoreCommentsFix` (really bad at naming) to the rule to ignore comments at all and always fix it. It'd be `false` by default, of course.